data/avro-schema/src/main/resources/avro/net/corda/data/chunking/Chunk.avsc

```diff
@@ -29,7 +32,10 @@
     },
     {
       "name": "offset",
-      "type": "long",
+      "type": [
+        "null",
+        "long"
+      ],
```

Comment:
Why does the offset need to be nullable?

Reply:
its specific to writing to a file. It has no use in the message bus integration or the flow session integration. I've a JIRA ticket to remove filename and offset and set them in the chunk property map instead.

Reply:
I understand the filename one, but isn't the offset just an indication of where the chunk is in the list of bytes? Isn't that completely generic for all chunking usecases?

Reply:
Ye I see your point. I haven't been tracking or using the bytes position in the message bus piece though, so it feels like it would be a bit of a no-op to set it.

Reply:
It's cheap to set as it's just an index that is available as you go through the stream when creating the chunks, and I suspect that it's always going to be more efficient to use this when putting the chunks back together on the other end, so I don't really see a downside in setting it.

Reply:
ive kept the changes minimal so far to the existing code. I've tried to avoid any refactoring the chunking stuff which sends chunks individually as it generates them and returns an id of the chunks. i dont bother with a stream as I need to avro serialize the data and it returns the full ByteArray. So then split it into chunk sized arrays. the offset will always be the bytearray size so i think i can set it based on that.

Reply:
For uses cases like saving chunked data to file, it would be good to have original data size as well, then we could save chunks directly to file as they come and considerably lower memory footprint for big files.

Reply:
@mbrkic-r3 we can't necessarily do this as we could be streaming chunks from the origin (e.g. the HTTP file upload). 
But I agree, if it is available, it could be put into the property map and used as an optimisation.